### PR TITLE
Minimize *SAN_OPTIONS for engine fuzzers

### PR DESCRIPTION
**Pull request by inferno-chromium**
Part 1 of fixing https://github.com/google/clusterfuzz/issues/353

### CODE REVIEW

**Comment by inferno-chromium** [comment]
this might slow down minimize untrusted test, i can fix it later before landing, but review should be ready.

File: src/python/bot/tasks/minimize_task.py [hunk 1]
```diff
@@ -1244,6 +1248,40 @@ def handle_unreproducible():
...
 
+  # Minimize *_OPTIONS env variable first.
+  env = {}
+  for tool in environment.SUPPORTED_MEMORY_TOOLS_FOR_OPTIONS:
+    options_env_var = tool + '_OPTIONS'
+    options = environment.get_memory_tool_options(options_env_var)
+    if not options:
+      continue
+
+    minimized_options = copy.deepcopy(options)
```
Comment (oliverchang) [inline src/python/bot/tasks/minimize_task.py]: nit: I think deepcopy is overkill here. We can just use options.copy() which performs a shallow copy.

File: src/python/bot/tasks/minimize_task.py [hunk 2]
```diff
@@ -1244,6 +1248,40 @@ def handle_unreproducible():
...
+    for option_name, option_var in six.iteritems(options):
+      minimized_options.pop(option_name)
+      environment.set_memory_tool_options(options_env_var, minimized_options)
+
+      _, crash_result = _run_libfuzzer_tool(
```
Comment (oliverchang) [inline src/python/bot/tasks/minimize_task.py]: should we be calling libfuzzer with minimize here? can we just run the testcase without minimizing (i.e. by calling _run_libfuzzer_testcase)? seems weird to do it here and then again from lines 1287 onwards.

File: src/python/bot/tasks/minimize_task.py (hunk 1 above)
Comment (oliverchang) [reply src/python/bot/tasks/minimize_task.py]: I think shallow copies should be sufficient here -- we don't need to make copies of the individual strings in the options dict.

**Pull request merged**